extensions/guacamole-auth-restrict/src/main/resources/controllers/timeRestrictionFieldController.js

```diff
@@ -0,0 +1,222 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
+/**
+ * Controller for time restriction fields, which are used to select weekday and
+ * time restrictions that apply to user logins and connections.
+ */
+angular.module('guacRestrict').controller('timeRestrictionFieldController', ['$scope', '$injector',
+        function timeRestrictionFieldController($scope, $injector) {
+    
+    /**
+     * Options which dictate the behavior of the input field model, as defined
+     * by https://docs.angularjs.org/api/ng/directive/ngModelOptions
+     *
+     * @type Object.<String, String>
+     */
+    $scope.modelOptions = {
+        
+        /**
+         * Space-delimited list of events on which the model will be updated.
+         *
+         * @type String
+         */
+        updateOn : 'blur',
+
+        /**
+         * The time zone to use when reading/writing the Date object of the
+         * model.
+         *
+         * @type String
+         */
+        timezone : 'UTC'
+
+    };
+    
+    // Required types
+    // const TimeRestrictionEntry = $injector.get('TimeRestrictionEntry');
+    
+    /**
+     * The restrictions, as objects, that are used by the HTML template to
+     * present the restrictions to the user via the web interface.
+     */
+    $scope.restrictions = [];
+
+    /**
+     * Map of weekday identifier to display name.
+     */
+    $scope.weekDays = [
+        { id : 1, day : 'Monday' },
+        { id : 2, day : 'Tuesday' },
+        { id : 3, day : 'Wednesday' },
+        { id : 4, day : 'Thursday' },
+        { id : 5, day : 'Friday' },
+        { id : 6, day : 'Saturday' },
+        { id : 7, day : 'Sunday' }
+    ];
+    
+    /**
+     * Remove the current entry from the list.
+     * 
+     * @param {type} entry
+     *     A restriction entry.
+     */
+    $scope.removeEntry = function removeEntry(entry) {
+        if (entry === null || entry.$$hashKey === '') {
+            return;
+        }
+        for (let i = 0; i < $scope.restrictions.length; i++) {
+            if ($scope.restrictions[i].$$hashKey === entry.$$hashKey) {
+                $scope.restrictions.splice(i,1);
+                return;
+            }
+        }
+    };
+    
+    /**
+     * Add an empty entry to the restriction list.
+     */
+    $scope.addEntry = function addEntry() {
+        $scope.restrictions.push({});
+    };
+    
+
+    /**
+     * Parse the provided string into an array containing the objects that
+     * represent each of entries that can then be displayed as a more
+     * user-friendly field.
+     * 
+     * @param {String} restrString
+     *     The string that contains the restrictions, un-parsed and as stored
+     *     in the underlying field.
+     *     
+     * @returns {Array|Object}
```

Comment:
This doesn't look correct - this is saying that the return type could be an array _or_ an object. An array of objects would be `Object[]`.

Reply:
Fixed via rebase.